paddle/fluid/framework/ir/conv_bn_fuse_pass.cc

```diff
@@ -151,26 +166,66 @@ std::unique_ptr<ir::Graph> ConvBNFusePass::ApplyImpl(
                                *bn_mean, *bn_variance, eltwise_y_in_tensor,
                                epsilon);
 
-    // Create an elementwise add node
-    OpDesc desc;
-    desc.SetInput("X", std::vector<std::string>({conv_out->Name()}));
-    desc.SetInput("Y", std::vector<std::string>({eltwise_y_in_node->Name()}));
-    desc.SetOutput("Out", std::vector<std::string>({bn_out->Name()}));
-    desc.SetType("elementwise_add");
-    desc.SetAttr("axis", 1);
-    bool a = boost::get<bool>(conv->Op()->GetAttr("use_mkldnn"));
-    desc.SetAttr("use_mkldnn", a);
-    auto eltwise_op = g->CreateOpNode(&desc);  // OpDesc will be copied.
-
-    GraphSafeRemoveNodes(graph.get(), {bn_scale, bn_bias, bn_mean, bn_variance,
-                                       batch_norm, bn_mean_out, bn_variance_out,
-                                       bn_saved_mean, bn_saved_variance});
-
-    PADDLE_ENFORCE(subgraph.count(conv_input));
-    IR_NODE_LINK_TO(conv_out, eltwise_op);
-    IR_NODE_LINK_TO(eltwise_y_in_node, eltwise_op);
-    IR_NODE_LINK_TO(eltwise_op, bn_out);
-
+    // check if MKL-DNN should be used
+    // TODO(wojtuss): replace after the "OP placement" pass is added
+    // bool use_mkldnn = boost::get<bool>(conv->Op()->GetAttr("use_mkldnn"));
+    bool use_mkldnn = IsMKLDNNSetOn();
+    // with MKL-DNN fuse conv+bn into conv with bias
+    // without MKL-DNN fuse conv+bn into conv+elementwise_add
+    if (use_mkldnn) {
+      auto conv_bias_names = conv->Op()->Input("Bias");
+      if (conv_bias_names.size() > 0) {
+        // reuse existing conv bias node
+        PADDLE_ENFORCE_EQ(conv_bias_names.size(), 1);
+        auto* conv_bias_var = scope->FindVar(conv_bias_names[0]);
+        auto* conv_bias_tensor = conv_bias_var->GetMutable<LoDTensor>();
+        PADDLE_ENFORCE_EQ(conv_bias_tensor->dims(),
+                          eltwise_y_in_tensor->dims());
+        using EigenVectorArrayMap =
+            Eigen::Map<Eigen::Array<float, Eigen::Dynamic, 1>>;
+        using ConstEigenVectorArrayMap =
+            Eigen::Map<const Eigen::Array<float, Eigen::Dynamic, 1>>;
+        EigenVectorArrayMap conv_bias_array(
+            conv_bias_tensor->mutable_data<float>(platform::CPUPlace()),
+            conv_bias_tensor->numel(), 1);
+        ConstEigenVectorArrayMap eltwise_y_in_array(
+            eltwise_y_in_tensor->mutable_data<float>(platform::CPUPlace()),
+            eltwise_y_in_tensor->numel(), 1);
+        conv_bias_array = conv_bias_array + eltwise_y_in_array;
```

Comment:
Maybe the part that does calculations with Eigen could (lines 184-194) could be moved to the separate function? It could make the code a bit more readable, I think.

Reply:
I reduced it to two lines